plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -215,6 +217,46 @@ public function getOptOutView()
         return $this->view;
     }
 
+    private function optOutStyling()
+    {
+        $cssfontsize = Common::getRequestVar('fontSize', false, 'string');
+        $cssfontcolour = Common::getRequestVar('fontColor', false, 'string');
+        $cssfontfamily = Common::getRequestVar('fontFamily', false, 'string');
+        $cssbackgroundcolor = Common::getRequestVar('backgroundColor', false, 'string');
+        $cssbody = 'body { ';
+
+        $hexstrings = array(
+            'fontColor' => $cssfontcolour, 
+            'backgroundColor' => $cssbackgroundcolor
+        );
+        foreach ($hexstrings as $key => $testcase) {
+            if ($testcase && !(ctype_xdigit($testcase) && in_array(strlen($testcase),array(3,6), true))) {
+                throw new \Exception("The URL parameter $key value of '$testcase' is not valid. Expected value is for example 'ffffff' or 'fff'.\n");
+            }
+        }
+
+        if ($cssfontsize && (preg_match("/^[0-9]+[\.]?[0-9]*(px|pt|em)$/", $cssfontsize))) {
+            $cssbody .= 'font-size: ' . $cssfontsize . '; '; 
+        } else if ($cssfontsize) {
+            throw new \Exception("The URL parameter fontSize value of '$cssfontsize' is not valid. Expected value is for example '15pt', '1.2em' or '13px'.\n");
+        }
+
+        if ($cssfontfamily && (preg_match("/^[a-zA-Z-\ ,]+$/", $cssfontfamily))) {
```

Comment:
This regex doesn't accept quotes (`"`) which are quite common as they are needed when the font has a space in it's name.
Example: `-apple-system,"Helvetica Neue Light",HelveticaNeue,"Helvetica Neue","Liberation Sans",Roboto,Arial,sans-serif`

Reply:
we checked and it seems browsers accept the font without double quotes as well... Should be OK

Reply:
But without quotes one needs to escape all spaces and special characters (according to https://drafts.csswg.org/css2/fonts.html#value-def-family-name)
And this breaks the feature for people who just paste the font-family from their website and then just see an exception.

Reply:
Added double quote in the accepted chars in: https://github.com/matomo-org/matomo/pull/12494